### PR TITLE
Auto-generate shades in colour palette & document Colors and Typography in Storybook

### LINKED ISSUE #3752: Create `Typography` & `Colors` -pages to `Styles` in Storybook
- [x] Introduce `Noto Sans` in a typography intro page.  Something similar to this:
https://master--5ccbc373887ca40020446347.chromatic.com/?path=/story/typography-base--page
(source: https://github.com/storybookjs/design-system/blob/master/src/components/Typography.stories.mdx)
- [x] Introduce colour system https://storybook.js.org/docs/react/writing-docs/doc-block-colorpalette

### LINKED ISSUE #3775: Styles / Colours
We have a color.ts file that documents our colours. 

A consideration is to use Darken and Lighten to product colours. To automatically create hover and active colours. (And better support theming in the future??)

This would allow us to allow config of our Primary colour for instance. So a country could change it and the hover and active states would reflect the new colour

This ticket is to assess and define our approach to colours

### PR DESCRIPTION
**Pull request by naftis**
Closes #3775
Closes #3752 

* `indigoDark` -> `primaryDark`
* `redDark` -> `negativeDark` 
* `greenDark` -> `positiveDark`
* ^ autogenerate dark shades for primary, negative, positive and teal. This is to help support theming in the future.
* Create documentation for colors and typography
* Installs `polished` for `lighten` / `darken` functions. **Question:** do we want to install a new dependency for this or to create our own functions? `polished` should support tree-shaking but [Bundlephobia](https://bundlephobia.com/package/polished@4.2.2) says that every export is the same size 🤔

### CODE REVIEW

File: packages/components/src/stories/colors.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    colors={{
+      opacity24: colors.opacity24,
+      opacity54: colors.opacity54
+    }}
+  />
+</ColorPalette>
+
+## Gradient
+
+? do we wanna include gradient here
```
Comment (naftis) [inline packages/components/src/stories/colors.stories.mdx:129]: todo

**Review by Zangetsu101 — COMMENTED** [review]
Looks good. Just wondering could we perhaps use `hsl` instead to create the lighter and darker shades?

**Comment by naftis** [comment]
@Zangetsu101 Good idea.

@jpye-finch what do you think, could HSL work for our color system? Then we could just adjust the lightness of the color. This would mean that theme colors would need to be in HSL in future installations too

**Comment by Zangetsu101** [comment]
It seems not all the colors are rendering properly.
image

**Comment by naftis** [comment]
@Zangetsu101 for me it flashed like that for a couple of seconds, but then started working properly, maybe the component library was still compiling on your end?
image

**Comment by Zangetsu101** [comment]
> @Zangetsu101 for me it flashed like that for a couple of seconds, but then started working properly, maybe the component library was still compiling on your end? Also fixed the issue with fonts not loading on the left side panel image

Alright. Then it's fine.

File: packages/components/.storybook/preview.tsx [hunk 2]
```diff
@@ -11,17 +11,10 @@
...
 import { getTheme } from '@opencrvs/components/lib/theme'
 
 const theme = getTheme()
 
-WebFont.load({
-  google: {
-    families: ['Noto+Sans:600', 'Noto+Sans:400']
-  }
-})
-
```
[the remark this answers is not in the document]
Comment (jpye-finch) [reply packages/components/.storybook/preview.tsx:22]: @Zangetsu101 good idea! Still working on trying to get Noto Sans uses in the Storybook canvas for Headings.  

but we should set ourselves up to support a localisation addon i18n in the future

**Review by Zangetsu101 — CHANGES_REQUESTED** [review]
The changes look good. It's just that the the color palette doesn't load properly if you run storybook with the `yarn storybook` command but it works with `yarn start`.

**Review by Zangetsu101 — APPROVED** [review]

**Pull request merged**